airflow/providers/amazon/aws/operators/rds.py

```diff
@@ -0,0 +1,605 @@
+    :param copy_tags: Whether to copy all tags from the source snapshot to the target snapshot (default False)
+    :type copy_tags: bool or None
+    :param pre_signed_url: The URL that contains a Signature Version 4 signed request
+    :type pre_signed_url: str or None
+    :param option_group_name: The name of an option group to associate with the copy of the snapshot
+        Only when db_type='instance'
+    :type option_group_name: str or None
+    :param target_custom_availability_zone: The external custom Availability Zone identifier for the target
+        Only when db_type='instance'
+    :type target_custom_availability_zone: str or None
+    :param source_region: The ID of the region that contains the snapshot to be copied
+    :type source_region: str or None
+    """
+
+    template_fields = (
+        "source_db_snapshot_identifier",
+        "target_db_snapshot_identifier",
+        "tags",
+        "pre_signed_url",
+        "option_group_name",
+    )
+
+    def __init__(
+        self,
+        *,
+        db_type: str,
+        source_db_snapshot_identifier: str,
+        target_db_snapshot_identifier: str,
+        kms_key_id: str = "",
+        tags: Optional[Sequence[TagTypeDef]] = None,
+        copy_tags: bool = False,
+        pre_signed_url: str = "",
+        option_group_name: str = "",
+        target_custom_availability_zone: str = "",
+        source_region: str = "",
+        aws_conn_id: str = "aws_default",
+        **kwargs,
+    ):
+        super().__init__(aws_conn_id=aws_conn_id, **kwargs)
+
+        self.db_type = RdsDbType(db_type)
+        self.source_db_snapshot_identifier = source_db_snapshot_identifier
+        self.target_db_snapshot_identifier = target_db_snapshot_identifier
+        self.kms_key_id = kms_key_id
+        self.tags = tags or []
+        self.copy_tags = copy_tags
+        self.pre_signed_url = pre_signed_url
+        self.option_group_name = option_group_name
+        self.target_custom_availability_zone = target_custom_availability_zone
+        self.source_region = source_region
+
+    def _describe_item(self, **kwargs) -> list:
+        """Returns existing snapshots"""
+        if self.db_type.value == "instance":
+            db_snapshots = self.hook.conn.describe_db_snapshots(
+                DBSnapshotIdentifier=self.target_db_snapshot_identifier, **kwargs
+            )
+            return db_snapshots['DBSnapshots']
+        else:
+            db_cluster_snapshots = self.hook.conn.describe_db_cluster_snapshots(
+                DBClusterSnapshotIdentifier=self.target_db_snapshot_identifier, **kwargs
+            )
+            return db_cluster_snapshots['DBClusterSnapshots']
+
+    def execute(self, context: 'Context') -> str:
+        self.log.info(
+            "Starting to copy snapshot '%s' as '%s'",
+            self.source_db_snapshot_identifier,
+            self.target_db_snapshot_identifier,
+        )
+
+        if self.db_type.value == "instance":
```

Comment:
this `if type == instance` decision everywhere creates a lot of code dupe.

i think there's got to be a way to simplify

one way that comes to mind is create a method

```python
def copy_snapshot(self, **kwargs)
    if 'instance':
        return self.client.copy_db_snapshot(**kwargs)
    else:
        return self.client.copy_db_cluster_snapshot(**kwargs)
```

or you could _possibly_ do something even more magical with getattr but magic has its costs...

but i recommend doing _something_ to reduce the code dupe throughout this PR

Reply:
The problem here is that each `boto3` method needs different arguments (, therefore this check should be implemented twice.
```
class RdsBaseOperator # or Mixin
    def copy_snapshot(self, **kwargs)
        if 'instance':
            return self.client.copy_db_snapshot(**kwargs)
        else:
            return self.client.copy_db_cluster_snapshot(**kwargs)
 ```
 and
 ```
 class RdsCopySnapshotOperator(RdsBaseOperator)
    def execute(self, **kwargs)
        if 'instance':
            return self.client.copy_snapshot(SourceDBSnapshotIdentifier=y, param2=x)
        else:
            return self.client.copy_snapshot(SourceDBClusterSnapshotIdentifier=x, param3=z)
```

Reply:
Yup didn't realize that before

Reply:
I implemented one solution. Seems it reduced some duplications